### PR TITLE
Add additional valueset extracts for other eICR Fields

### LINKED ISSUE #99: Add extract methods for the other data elements

### PR DESCRIPTION
**Pull request by BradySkylight**
## Description

This PR tackles the code to be able to run the process to get that data from various valuesets for eICR fields as well as the actual data extract files (from running the process).  This PR completes this for the following additional fields:

| **eICR Field** | **Value Set** |
| -------- | ------- |
| Encompassing Encounter Code (Encounter Type) | HL7 Act Encounter Codes |
| Medications Administered (list) | RXNORM Medication Codes |
| Immunization Vaccine Code | CVX Immunization Codes |
| Diagnosis / Symptom Code | SNOMED Problem Codes |

This PR also includes updates to the README and subsequent images in the data_curation folder.

## Related Issues

Closes #99 
## Additional Notes

For more information, reference the document and notes in the table documented here: https://docs.google.com/document/d/1ZMv0y21n1FsSOCF40NX03VPF9leU-gHnOW_fZCAqur8/edit?tab=t.0#heading=h.h6l3qgjh8lv4

<--------------------- REMOVE THE LINES BELOW BEFORE MERGING --------------------->

## Checklist

Please review and complete the following checklist before submitting your pull request:

- [X] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [X] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [X] I have updated the documentation, if applicable.
- [X] I have added or updated test cases to cover my changes, if applicable.
- [X] I have minimized the number of reviewers to include only those essential for the review.

## Checklist for Reviewers

Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: data_curation/terminology_valueset_sync.py [hunk 1]
```diff
@@ -685,6 +691,129 @@ def _filter_loinc_term(text: str) -> bool:
...
+                        ).strip()
+                    else:
+                        result_row["description"] = ""
+
+                    hl7_rows.append(result_row)
+        # Hard coded external encounter code that 'SHOULD' be used if it is an External Encounter
+        external_encounter = {
+            "code": "PHC2237",
+            "text": "External Encounter",
+            "description": "External Encounter",
```
[the remark this answers is not in the document]
Comment (BradySkylight) [reply data_curation/terminology_valueset_sync.py]: This is the specified code, based upon the eICR specificiation, if an encounter is not associated with a specific 'patient visit'.  You use the PHC2237 code for "External Encounter" in an eICR when a public health trigger occurs outside of a specific patient encounter, meaning it is not related to a particular visit or hospitalization.

https://docs.google.com/document/d/1ZMv0y21n1FsSOCF40NX03VPF9leU-gHnOW_fZCAqur8/edit?tab=t.0#heading=h.urxp8o8al6km
